test/codegen/bloom_filter_test.cpp

```diff
@@ -0,0 +1,360 @@
+    InsertTuple(vals, table1, txn);
+
+    curr_size += table1_tuple_size;
+  }
+
+  LOG_INFO("Finish populating test1");
+
+  // Load the inner table which contains twice tuples as the outer table
+  auto *table2 = catalog->GetTableWithName(DEFAULT_DB_NAME, table2_name, txn);
+  unsigned outer_table_cardinality = numbers.size() * outer_to_inner_ratio;
+  for (unsigned i = 0; i < outer_table_cardinality; i++) {
+    int number;
+    if (rand() % 100 < selectivity * 100) {
+      // Pick a random number from the inner table
+      number = numbers[rand() % numbers.size()];
+    } else {
+      // Pick a random number that is not in inner table
+      do {
+        number = rand();
+      } while (number_set.count(number) == 1);
+    }
+    std::vector<int> vals(UpDivide(table2_tuple_size, bigint_size), number);
+    InsertTuple(vals, table2, txn);
+  }
+
+  LOG_INFO("Finish populating test2\n");
+
+  // Build and execute the join plan
+  int num_iter = 3;
+  std::string query =
+      "SELECT * FROM test1 as t1, test2 as t2 "
+      "WHERE t1.c0 = t2.c0";
+
+  // Execute plan with bloom filter disabled
+  LOG_INFO("Executing without bloom filter");
+  double runtime1 = ExecuteJoin(query, txn, num_iter, numbers.size(), false);
+
+  // Execute plan with bloom filter enabled
+  LOG_INFO("\n");
+  LOG_INFO("Executing with bloom filter");
+  double runtime2 = ExecuteJoin(query, txn, num_iter, numbers.size(), true);
+
+  LOG_INFO("Avg With Bloom Filter Disabled: %f", runtime1);
+  LOG_INFO("Avg With Bloom Filter Enabled: %f", runtime2);
+  LOG_INFO("Ratio: %f", runtime2 / runtime1);
+
+  txn_manager.CommitTransaction(txn);
+}
+
+double BloomFilterCodegenTest::ExecuteJoin(std::string query,
+                                           concurrency::Transaction *txn,
+                                           int num_iter,
+                                           unsigned inner_table_cardinality,
+                                           bool enable_bloom_filter) {
+  std::unique_ptr<optimizer::AbstractOptimizer> optimizer(
+      new optimizer::Optimizer());
+  double total_runtime = 0;
+  // Run hash join multiple times and calculate the average runtime
+  for (int i = 0; i < num_iter; i++) {
+    auto plan =
+        TestingSQLUtil::GeneratePlanWithOptimizer(optimizer, query, txn);
+    assert(((planner::SeqScanPlan *)plan->GetChild(0))->GetTable()->GetName() ==
+           table1_name);
+    // Change the bloom filter flag and set the correct cardinality in the plan
+    const_cast<planner::AbstractPlan *>(plan->GetChild(0))
+        ->SetCardinality(inner_table_cardinality);
+    dynamic_cast<planner::HashJoinPlan *>(plan.get())
+        ->SetBloomFilterFlag(enable_bloom_filter);
+    double runtime = CompileAndExecute(*plan, txn);
+
+    LOG_INFO("Execution Time: %0.0f ms", runtime);
+    total_runtime += runtime;
+  }
+  return total_runtime / num_iter;
+}
+
+double BloomFilterCodegenTest::CompileAndExecute(
```

Comment:
There is a `CompileAndExecute` in `PelotonCodegenTestUtils` that essentially does the same thing. Can you merge this function and that one? IIRC, that one doesn't return `RuntimeStats` which you need.